src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PropertySubPatternCompletionProviderTests.cs

```diff
@@ -0,0 +1,655 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    [Trait(Traits.Feature, Traits.Features.Completion)]
+    public class PropertySubpatternCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public PropertySubpatternCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider()
+        {
+            return new PropertySubpatternCompletionProvider();
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { $$ }
+    }
+}
+";
+            // VerifyItemExistsAsync also tests with the item typed.
+            await VerifyItemExistsAsync(markup, "P1");
+            await VerifyItemExistsAsync(markup, "P2");
```

Comment:
do these test when nothing after the $$ is in the file?  Or do we need specific tests for that?

Reply:
@CyrusNajmabadi they do not (I was curious about making them do that once)